pallets/xcm-transactor/src/lib.rs

```diff
@@ -1167,8 +1166,9 @@ pub mod pallet {
 
 		/// Ensure `dest` has chain part and none recipient part.
 		fn ensure_valid_dest(dest: &Location) -> Result<Location, DispatchError> {
-			if let (Some(dest), None) = (dest.chain_part(), dest.non_chain_part()) {
-				Ok(dest)
+			let chain_location = dest.chain_location();
+			if *dest == chain_location {
+				Ok(chain_location)
```

Comment:
I find the previous code easier to understand, this change is confusing to me

Reply:
`chain_part` and `non_chain_part` are defined in orml (in the `Parse` trait), so we can no longer use them

Reply:
I agree with Rodri, at least we should update the comment to reflect the new impl:
"/// Ensure `dest` has chain part and none recipient part."